### ISSUE TITLE
Need linux version of my plugins

### ISSUE THREAD

**Issue by elanhickler** [RobinSchmidt/RS-MET#26]
So you made a linux version of Chaosfly I think. You had some trouble with it? It seems like you figured it out well enough. Could you make more linux versions of my plugins?

**Comment by RobinSchmidt** [comment]
i had trouble with the codeblocks project produced by the projucer (at the end, i got it to compile, but the plugin crashed). with the makefile, everything was fine. you just add a "Linux Makefile" build target to the projucer project and then you open a terminal window inside this folder and type "make" or "make CONFIG=Release" for a release build. i can do this for you, when you tell me which version i should build but i guess at some point, you want to install a linux vm (as you have the vmware already up and running anyway, this should be easy) ....ahh...but wait....you need to also install some developer packages (libraries and headers), otherwise, you'll get compiler and/or linker errors. but i think, the error messages will tell you, what to install.

**Comment by elanhickler** [comment]
What VST host do you recommend for linux?

**Comment by RobinSchmidt** [comment]
for test/debug, i just use the juce vst-host (also on windows - and should i go mac, i guess i will use it there, too)

**Comment by elanhickler** [comment]
I have a bunch of "No package found, perhaps you should add the directory containing ..."

How do you deal with this? I know you have to use the terminal commands, what is the command?

**Comment by RobinSchmidt** [comment]
yes yes - that's what i meant when saying: you may have to install some developer packages. i usually do this with the software center - no need to deal with the command line terminal. you need to figure out the names of the missing packages, then enter that name into the "search" field of the software center, and when it was found, click on "install"

**Comment by RobinSchmidt** [comment]
this is a little bit of a tedious process. i wish, i would have taken notes (i.e. bookkeeping all the names of the packages) to make it easier next time. alas, i didn't keep / note down all the package names. maybe you can do this while installing?

**Comment by RobinSchmidt** [comment]
because, iirc, sometimes it was none-obvious what was missing and i needed some googling to figure it out :-(

**Comment by elanhickler** [comment]
No clue where to start. I searched in software center, I searched google, can't figure out the first step.

```
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

elan@elan-VirtualBox:~/Desktop/JUCE/extras/Projucer/Builds/LinuxMakefile$ make CONFIG=Release
Package freetype2 was not found in the pkg-config search path.
Perhaps you should add the directory containing `freetype2.pc'
to the PKG_CONFIG_PATH environment variable
No package 'freetype2' found
Package libcurl was not found in the pkg-config search path.
Perhaps you should add the directory containing `libcurl.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libcurl' found
Package x11 was not found in the pkg-config search path.
Perhaps you should add the directory containing `x11.pc'
to the PKG_CONFIG_PATH environment variable
No package 'x11' found
Package xext was not found in the pkg-config search path.
Perhaps you should add the directory containing `xext.pc'
to the PKG_CONFIG_PATH environment variable
No package 'xext' found
Package xinerama was not found in the pkg-config search path.
Perhaps you should add the directory containing `xinerama.pc'
to the PKG_CONFIG_PATH environment variable
No package 'xinerama' found
Package webkit2gtk-4.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `webkit2gtk-4.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'webkit2gtk-4.0' found
Package gtk+-x11-3.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gtk+-x11-3.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gtk+-x11-3.0' found
Package freetype2 was not found in the pkg-config search path.
Perhaps you should add the directory containing `freetype2.pc'
to the PKG_CONFIG_PATH environment variable
No package 'freetype2' found
Package libcurl was not found in the pkg-config search path.
Perhaps you should add the directory containing `libcurl.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libcurl' found
Package x11 was not found in the pkg-config search path.
Perhaps you should add the directory containing `x11.pc'
to the PKG_CONFIG_PATH environment variable
No package 'x11' found
Package xext was not found in the pkg-config search path.
Perhaps you should add the directory containing `xext.pc'
to the PKG_CONFIG_PATH environment variable
No package 'xext' found
Package xinerama was not found in the pkg-config search path.
Perhaps you should add the directory containing `xinerama.pc'
to the PKG_CONFIG_PATH environment variable
No package 'xinerama' found
Makefile:422: recipe for target 'check-pkg-config' failed
make: *** [check-pkg-config] Error 1
elan@elan-VirtualBox:~/Desktop/JUCE/extras/Projucer/Builds/LinuxMakefile$
```

**Comment by elanhickler** [comment]
Would like a linux version of Spiral Generator 2, please build one as soon as you can or explain more fully how to solve my missing package issues. Thanks!

**Comment by elanhickler** [comment]
ok i just installed the needed libraries and I got pretty far into getting a linux build of SpiralGenerator but got stuck because of an error coming from rosic

If you can get it to build ONCE then I think I can take over from there and build it whenever I need to. Update the jucer and whatever else is needed and push changes to repository.

**Comment by RobinSchmidt** [comment]
update my library and try again. seems like i introduced some build issues on linux when restructuring the includes for the rosic module. i fixed them now

**Comment by elanhickler** [comment]
missing juce files errors, I don't know what I'm doing wrong, did you try building Spiral Generator?

**Comment by RobinSchmidt** [comment]
aaahhh! yes.... it is not enough to just add the linux makefile target. you also have to set some search header paths and link to libraries. i now updated your .jucer file. i point to the modules folders within my rs-met directory. you may (or may not) want to change that, if you want to build against your newer juce version (but probably best to leave it as is, if it works)

**Comment by RobinSchmidt** [comment]
did you take notes of the names of the packages to install? because i'm currently in the process of doing all that again on my new pc (have set up a linux vm on it)...and i don't find them in the software center anymore ...had to hand install via the terminal like:
```
sudo apt-get install libasound2-dev 
sudo apt-get install libfreetype6-dev
```
...but i need some more and don't know the names. i probably can figure out one by one using google, but if you have them, that would make it easier. linux sucks! :-(

**Comment by RobinSchmidt** [comment]
nevermind - figured it out. it is
```
sudo apt-get install libasound2-dev 
sudo apt-get install libfreetype6-dev
sudo apt-get install libx11-dev
sudo apt-get install libxrandr-dev 
sudo apt-get install libxinerama-dev
sudo apt-get install libxcursor-dev
```
i also added this information to a textfile in my repo - for reference for the next time anyone has to do this
